src/edu/wright/cs/raiderplanner/controller/SettingsController.java

```diff
@@ -0,0 +1,334 @@
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Color;
+import javafx.scene.text.Font;
+import javafx.scene.text.FontPosture;
+import javafx.scene.text.FontWeight;
+import javafx.scene.text.TextAlignment;
+import javafx.stage.Screen;
+import javafx.util.Duration;
+
+import java.net.URL;
+import java.util.ResourceBundle;
+
+/**
+ * Actions associated with the settings menu and its items.
+ * Settings menu implemented in correspondence with Issue #238.
+ * 2/1/2018 - Template Created
+ * @author Clayton D. Terrill
+ */
+public class SettingsController implements Initializable {
+
+	/**
+	 * Initializes switch names and other buttons.
+	 * Represent states that the settings menu can be in.
+	 * May be assigned to a button in the initialize method and
+	 * what it does may be applied in the main method's switch statement.
+	 */
+	public enum Window {
+		EMPTY, ABOUT, GENERAL, ACCOUNT, THEME, NOTIFICATIONS
+	}
+
+	private Window current;
+	private boolean isNavOpen;
+
+	// Screen size:
+	private double screenWidth = Screen.getPrimary().getVisualBounds().getWidth();
+	private double screenHeight = Screen.getPrimary().getVisualBounds().getHeight();
+	private double screenAverage = (screenWidth + screenHeight) / 2;
+
+	// Shadows:
+	private int navShadowRadius = (int) (screenAverage * 0.03);
+	private int navShadowOffset = (int) (screenAverage * 0.01);
+	private DropShadow navShadow = new DropShadow(navShadowRadius, navShadowOffset, 0, Color.BLACK);
+	// private DropShadow notifShadow = new DropShadow(screenAverage * 0.02, 0, 0.009, Color.BLACK);
+
+	// Labels:
+	private Label welcome;
+	@FXML
+	private Label title;
+
+	// Buttons:
+	@FXML
+	private Button openMenu;
+	@FXML
+	private Button generalBtn;
+	@FXML
+	private Button accountBtn;
+	@FXML
+	private Button themeBtn;
+	@FXML
+	private Button notificationsBtn;
+	@FXML
+	private Button aboutBtn;
+	@FXML
+	private Button closeDrawer;
+
+	// Panes:
+	@FXML
+	private AnchorPane navList;
+	@FXML
+	private GridPane mainContent;
+	@FXML
+	private HBox topBox;
+	@FXML
+
+	// chat variables
+	private final BorderPane mainPane = new BorderPane();
+
+	/**
+	 * Sets this.current to equal passed variable and calls this.main().
+	 */
+	public void main(Window wind) {
+		this.current = wind;
+		this.main();
+	}
+
+	/**
+	 * Main method containing switch statements.
+	 * Switch statements detect what state the app is in.
+	 */
+	public void main() {
+		if (isNavOpen) {
+			openMenu.fire();
+		}
+
+		switch (this.current) {
+		case ABOUT: {
+			loadAbout();
+			break;
+		}
+		case GENERAL: {
+			this.loadGeneral();
+			break;
+		}
+		case ACCOUNT: {
+			this.loadAccount();
+			break;
+		}
+		case THEME: {
+			this.loadTheme();
+			break;
+		}
+		case NOTIFICATIONS: {
+			this.loadNotification();
+			break;
+		}
+		default:
+			break;
+		}
+	}
+
+	/**
+	 * Display the About menu.
+	 */
+	public void loadAbout() {
```

Comment:
I'm not too sure, but most of this function looks like it's changing and adjusting the `view`, rather than being the controller. I think the same might be able to be said for all of the `@FMXL` declarations.

The view should be the one to decide how to display the information, text, layout, etc., the model should store the data, and the controller should be what communicates between the two.

I would check out this link for more information:

https://www.tutorialspoint.com/design_pattern/mvc_pattern.htm

I may be wrong though. I'm not an expert in MVC.

Reply:
To remain consistent, I implemented the SettingsController the same as the MainController. The idea is that individual settings will have their own Java class that may be called using the SettingsController.